connect/runtime/src/main/java/org/apache/kafka/connect/runtime/TransformationChain.java

```diff
@@ -68,4 +74,11 @@ public int hashCode() {
         return Objects.hash(transformations);
     }
 
+    public String toString() {
+        StringJoiner str = new StringJoiner(", ", "<" + getClass().getSimpleName() + ": ", ">");
+        for (final Transformation<R> transformation : transformations) {
```

Comment:
I'm against the sprinkling of `final` keyword everywhere. But more than that, I'm in favor of consistency w.r.t. the surrounding code. A few lines above, you may observe that `close` has a similar loop without `final`. Every project has its idiosyncrasies and Connect is not dogmatic w.r.t to `final` for local variables.

Reply:
Also, am I correct to read that every transformation name will be prefixed by this classes simple classname as well. Is this the visual result we want the most? Can we get an example as a non-javadoc comment here, if it's useful?

Reply:
I agree with consistency re: `final`. 

I'm not sure where your confusion re:simpleName is coming from. I hesitate to put an example of the output in the comments since it could lead to future inconsistency, but the intention is to put simple name of each transform with the record it will be applied to. I will paste a sample log output below

Reply:
[2018-11-13 16:06:08,083] TRACE Registered Applying Single Message Transformation (SMT) HoistField to SinkRecord{kafkaOffset=45, timestampType=NoTimestampType} ConnectRecord{topic='test', kafkaPartition=12, key=null, keySchema=null, value={a=10}, valueSchema=null, timestamp=null, headers=ConnectHeaders(headers=)} (org.apache.kafka.connect.runtime.TransformationChain:47)

Reply:
I don't feel too strong about the comment above the code, even though I don't think there's an issue of that becoming stale. It's what this method does and should be updated when (infrequently) this method changes, if ever. 

But in your example above, I don't see neither the prefix nor the suffix that are given to the `StringJoiner`

Reply:
@cyrusv 
Your example should print something like: 
`<SimpleClassName: blah blah blah>`
But I don't see `<` or `>` in your example. So either the code example is wrong, or you didn't mean to use those "quotes" in the joiner. BTW probably not adding this decoration would be better.

Reply:
I see your point -- I updated the code to match the style used in `ConnectRecord.toString()`, so now the output will look like "TransformationChain{FaultyPassthrough, FaultyPassthrough}", as verified by running the unit tests.